### DISCUSSION TITLE
Foreign Key as array element

### DISCUSSION THREAD

**Discussion by tawjaw** [supabase/supabase#781] [Questions] [answered]
Hello,

Is it possible to have array elements reference an id from another table like a foreign key? 

For example table products with product_id

And table orders
With 
items integer[] 

Where elements inside items array are referencing product_id from products.

I know in this example following a one-many relational design works better. 

In my use case order matters. In this example it would be the order of items inside items[]
And instead of adding order index into the relationship and updating all the order index whenever there's an update.. using an array looks like the right way to go.

I can still use array and store ids without references but it means I have to manage the constraints myself with triggers.

**Comment by steve-chavez** [comment, marked as answer]
Hey @tawjaw,

This was asked before, please check https://github.com/supabase/supabase/discussions/535

**Reply by tawjaw** [reply]
Thanks!
